# Week 1 Coding Tasks
Sharice Clough

1. Import Pandas library

In [1]:
import pandas as pd

2. Read in the data (Hospitals.csv).
This datset was downloaded from [the Homeland Infrastructure Foundation-Level Data (HIFLD) database](https://hifld-geoplatform.opendata.arcgis.com/datasets/hospitals/explore).

In [28]:
hospitals = pd.read_csv("../data/Hospitals.csv")

3. Look at the first few rows and then look at the last few.

In [29]:
hospitals.head(6)

X             Y  OBJECTID        ID  \
0 -1.331889e+07  4.346975e+06         1   5793230   
1 -1.322651e+07  4.049626e+06         2  53391362   
2 -1.315620e+07  4.031978e+06         3  11190023   
3 -1.317190e+07  4.041752e+06         4  17090028   
4 -1.313208e+07  4.037270e+06         5  23691706   
5 -1.315222e+07  4.009980e+06         6  25190712   

                                                NAME  \
0                    CENTRAL VALLEY GENERAL HOSPITAL   
1  LOS ROBLES HOSPITAL & MEDICAL CENTER - EAST CA...   
2                  EAST LOS ANGELES DOCTORS HOSPITAL   
3          SOUTHERN CALIFORNIA HOSPITAL AT HOLLYWOOD   
4                      KINDRED HOSPITAL BALDWIN PARK   
5                   LAKEWOOD REGIONAL MEDICAL CENTER   

                     ADDRESS             CITY STATE    ZIP           ZIP4  \
0    1025 NORTH DOUTY STREET          HANFORD    CA  93230  NOT AVAILABLE   
1             150 VIA MERIDA  WESTLAKE VILAGE    CA  91362  NOT AVAILABLE   
2    4060 WHITTIER BOULEVARD      LOS ANGELES    CA  90023  NOT AVAILABLE   
3     6245 DE LONGPRE AVENUE        HOLLYWOOD    CA  90028  NOT AVAILABLE   
4  14148 FRANCISQUITO AVENUE     BALDWIN PARK    CA  91706  NOT AVAILABLE   
5     3700 EAST SOUTH STREET         LAKEWOOD    CA  90712  NOT AVAILABLE   

   ...             VAL_DATE                           WEBSITE       STATE_ID  \
0  ...  2014/02/10 00:00:00      http://www.hanfordhealth.com  NOT AVAILABLE   
1  ...  2014/02/10 00:00:00  http://www.losrobleshospital.com  NOT AVAILABLE   
2  ...  2014/02/10 00:00:00             http://www.elalax.com  NOT AVAILABLE   
3  ...  2014/02/10 00:00:00         http://sch-hollywood.com/  NOT AVAILABLE   
4  ...  2014/02/10 00:00:00      http://www.khbaldwinpark.com  NOT AVAILABLE   
5  ...  2014/02/10 00:00:00   http://www.lakewoodregional.com  NOT AVAILABLE   

                                    ALT_NAME ST_FIPS        OWNER TTL_STAFF  \
0                              NOT AVAILABLE       6  PROPRIETARY      -999   
1                              NOT AVAILABLE       6  PROPRIETARY      -999   
2                              NOT AVAILABLE       6  PROPRIETARY      -999   
3  HOLLYWOOD COMMUNITY HOSPITAL OF HOLLYWOOD       6  PROPRIETARY      -999   
4                              NOT AVAILABLE       6  PROPRIETARY      -999   
5                              NOT AVAILABLE       6  PROPRIETARY      -999   

   BEDS         TRAUMA  HELIPAD  
0    49  NOT AVAILABLE        N  
1    62  NOT AVAILABLE        N  
2   127  NOT AVAILABLE        N  
3   100  NOT AVAILABLE        N  
4    95  NOT AVAILABLE        N  
5   172  NOT AVAILABLE        N  

[6 rows x 34 columns]

In [30]:
hospitals.tail(6)

X             Y  OBJECTID         ID  \
7590 -9.819309e+06  4.136987e+06      7606  185838852   
7591 -8.566905e+06  4.728815e+06      7607  183520904   
7592 -1.219624e+07  3.676931e+06      7608  192185607   
7593 -8.084677e+06  5.094112e+06      7609  196706457   
7594 -8.522423e+06  4.718512e+06      7610  191021401   
7595 -1.244921e+07  4.976468e+06      7611  196784108   

                                               NAME  \
7590                 NORTH MS MEDICAL CENTER - IUKA   
7591  ADVENTIST HEALTHCARE WHITE OAK MEDICAL CENTER   
7592      COPPER QUEEN DOUGLAS EMERGENCY DEPARTMENT   
7593                      WHITING FORENSIC HOSPITAL   
7594             J KENT MCNEW FAMILY MEDICAL CENTER   
7595                  UNIVERSITY ORTHOPAEDIC CENTER   

                         ADDRESS            CITY STATE    ZIP           ZIP4  \
7590            177 CURTIS DRIVE            IUKA    MS  38852  NOT AVAILABLE   
7591           11890 HEALING WAY   SILVER SPRING    MD  20904  NOT AVAILABLE   
7592           100 E. 5TH STREET         DOUGLAS    AZ  85607  NOT AVAILABLE   
7593                70 OBRIEN DR      MIDDLETOWN    CT   6457           3945   
7594  175 HARRY S TRUMAN PARKWAY       ANNAPOLIS    MD  21401  NOT AVAILABLE   
7595              590 WAKARA WAY  SALT LAKE CITY    UT  84108  NOT AVAILABLE   

      ...             VAL_DATE  \
7590  ...  2020/05/05 00:00:00   
7591  ...  2020/05/05 00:00:00   
7592  ...  2020/05/05 00:00:00   
7593  ...  2020/05/08 00:00:00   
7594  ...  2020/05/05 00:00:00   
7595  ...  2020/05/05 00:00:00   

                                                WEBSITE       STATE_ID  \
7590  https://www.nmhs.net/locations/profile/north-m...  NOT AVAILABLE   
7591  https://www.adventisthealthcare.com/locations/...  NOT AVAILABLE   
7592  https://cqch.org/cqch-clinics/copper-queen-dou...  NOT AVAILABLE   
7593  https://portal.ct.gov/dmhas/wfh/whiting-forens...  NOT AVAILABLE   
7594  https://aahs.org/locations/j--kent-mcnew-famil...  NOT AVAILABLE   
7595          https://healthcare.utah.edu/orthopaedics/  NOT AVAILABLE   

           ALT_NAME ST_FIPS               OWNER TTL_STAFF  BEDS  \
7590  NOT AVAILABLE      28          NON-PROFIT      -999    48   
7591  NOT AVAILABLE      24          NON-PROFIT      -999   178   
7592  NOT AVAILABLE       4          NON-PROFIT      -999  -999   
7593  NOT AVAILABLE       9  GOVERNMENT - STATE      -999   229   
7594  NOT AVAILABLE      24          NON-PROFIT      -999  -999   
7595  NOT AVAILABLE      49  GOVERNMENT - STATE      -999  -999   

             TRAUMA  HELIPAD  
7590       LEVEL IV        Y  
7591  NOT AVAILABLE        N  
7592       LEVEL IV        N  
7593  NOT AVAILABLE        N  
7594  NOT AVAILABLE        N  
7595  NOT AVAILABLE        N  

[6 rows x 34 columns]

There are 7596 rows and 34 columns. Confirm with shape() function.

In [31]:
hospitals.shape

(7596, 34)

4. We don't need all the columns. Slice the dataframe so that it contains only the following columns: "NAME", "ADDRESS", "CITY", "STATE", "TYPE", "POPULATION", "COUNTY", "COUNTYFIPS", "NAICS_DESC", "BEDS"

In [32]:
hospitals = hospitals[["NAME", "ADDRESS", "CITY", "STATE", "TYPE", "POPULATION", "COUNTY", "COUNTYFIPS", "NAICS_DESC", "BEDS"]]

In [33]:
hospitals.head(6)

NAME  \
0                    CENTRAL VALLEY GENERAL HOSPITAL   
1  LOS ROBLES HOSPITAL & MEDICAL CENTER - EAST CA...   
2                  EAST LOS ANGELES DOCTORS HOSPITAL   
3          SOUTHERN CALIFORNIA HOSPITAL AT HOLLYWOOD   
4                      KINDRED HOSPITAL BALDWIN PARK   
5                   LAKEWOOD REGIONAL MEDICAL CENTER   

                     ADDRESS             CITY STATE                TYPE  \
0    1025 NORTH DOUTY STREET          HANFORD    CA  GENERAL ACUTE CARE   
1             150 VIA MERIDA  WESTLAKE VILAGE    CA  GENERAL ACUTE CARE   
2    4060 WHITTIER BOULEVARD      LOS ANGELES    CA  GENERAL ACUTE CARE   
3     6245 DE LONGPRE AVENUE        HOLLYWOOD    CA  GENERAL ACUTE CARE   
4  14148 FRANCISQUITO AVENUE     BALDWIN PARK    CA  GENERAL ACUTE CARE   
5     3700 EAST SOUTH STREET         LAKEWOOD    CA  GENERAL ACUTE CARE   

   POPULATION       COUNTY COUNTYFIPS                              NAICS_DESC  \
0          49        KINGS      06031  GENERAL MEDICAL AND SURGICAL HOSPITALS   
1          62      VENTURA      06111  GENERAL MEDICAL AND SURGICAL HOSPITALS   
2         127  LOS ANGELES      06037  GENERAL MEDICAL AND SURGICAL HOSPITALS   
3         100  LOS ANGELES      06037  GENERAL MEDICAL AND SURGICAL HOSPITALS   
4          95  LOS ANGELES      06037  GENERAL MEDICAL AND SURGICAL HOSPITALS   
5         172  LOS ANGELES      06037  GENERAL MEDICAL AND SURGICAL HOSPITALS   

   BEDS  
0    49  
1    62  
2   127  
3   100  
4    95  
5   172

5. What are the different hospital types (contained in the "TYPE" column)? Which is the most common?

In [34]:
hospitals["TYPE"].unique()

array(['GENERAL ACUTE CARE', 'PSYCHIATRIC', 'CHILDREN', 'LONG TERM CARE',
       'CRITICAL ACCESS', 'REHABILITATION', 'MILITARY', 'WOMEN',
       'SPECIAL', 'CHRONIC DISEASE'], dtype=object)

In [35]:
hospitals["TYPE"].value_counts()

GENERAL ACUTE CARE    4481
CRITICAL ACCESS       1027
PSYCHIATRIC            759
LONG TERM CARE         431
REHABILITATION         378
MILITARY               216
SPECIAL                145
CHILDREN               130
WOMEN                   20
CHRONIC DISEASE          9
Name: TYPE, dtype: int64

6. Find all hospitals in Nashville, Tennessee. Take a look at the BEDS column. What do you notice? Why do you think this is? (-999 used as missing value placeholder)

In [36]:
TN_hospitals = hospitals.loc[hospitals["STATE"] == "TN"]

In [37]:
TN_hospitals.loc[TN_hospitals["CITY"] == "NASHVILLE"]

NAME  \
545                  TRISTAR SKYLINE MEDICAL CENTER   
3449          SELECT SPECIALTY HOSPITAL - NASHVILLE   
3485                           CURAHEALTH NASHVILLE   
5429             THE WOMEN'S HOSPITAL AT CENTENNIAL   
5433        METROPOLITAN NASHVILLE GENERAL HOSPITAL   
6289                  SAINT THOMAS MIDTOWN HOSPITAL   
6293  VANDERBILT STALLWORTH REHABILITATION HOSPITAL   
6661           VA MEDICAL CENTER - NASHVILLE CAMPUS   
6774    SAINT THOMAS HOSPITAL FOR SPECIALTY SURGERY   
6777                     SAINT THOMAS WEST HOSPITAL   
6786          TRISTAR SOUTHERN HILLS MEDICAL CENTER   
6808              TRISTAR CENTENNIAL MEDICAL CENTER   
6813                 VANDERBILT UNIVERSITY HOSPITAL   

                             ADDRESS       CITY STATE                TYPE  \
545              3441 DICKERSON PIKE  NASHVILLE    TN  GENERAL ACUTE CARE   
3449    2000 HAYES STREET, SUITE 500  NASHVILLE    TN      LONG TERM CARE   
3485  1412 COUNTY HOSPITAL ROAD, B-1  NASHVILLE    TN      LONG TERM CARE   
5429              2221 MURPHY AVENUE  NASHVILLE    TN               WOMEN   
5433              1818 ALBION STREET  NASHVILLE    TN  GENERAL ACUTE CARE   
6289              2000 CHURCH STREET  NASHVILLE    TN  GENERAL ACUTE CARE   
6293             2201 CHILDREN'S WAY  NASHVILLE    TN      REHABILITATION   
6661                 1310 24TH AVE S  NASHVILLE    TN            MILITARY   
6774              2011 MURPHY AVENUE  NASHVILLE    TN  GENERAL ACUTE CARE   
6777               4220 HARDING ROAD  NASHVILLE    TN  GENERAL ACUTE CARE   
6786                391 WALLACE ROAD  NASHVILLE    TN  GENERAL ACUTE CARE   
6808           2300 PATTERSON STREET  NASHVILLE    TN  GENERAL ACUTE CARE   
6813       1211 MEDICAL CENTER DRIVE  NASHVILLE    TN  GENERAL ACUTE CARE   

      POPULATION    COUNTY COUNTYFIPS  \
545          213  DAVIDSON      47037   
3449          70  DAVIDSON      47037   
3485          60  DAVIDSON      47037   
5429        -999  DAVIDSON      47037   
5433         150  DAVIDSON      47037   
6289         683  DAVIDSON      47037   
6293          80  DAVIDSON      47037   
6661         493  DAVIDSON      47037   
6774          23  DAVIDSON      47037   
6777         541  DAVIDSON      47037   
6786         126  DAVIDSON      47037   
6808         733  DAVIDSON      47037   
6813        1091  DAVIDSON      47037   

                                             NAICS_DESC  BEDS  
545              GENERAL MEDICAL AND SURGICAL HOSPITALS   213  
3449  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE AB...    70  
3485  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE AB...    60  
5429  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE AB...  -999  
5433             GENERAL MEDICAL AND SURGICAL HOSPITALS   150  
6289             GENERAL MEDICAL AND SURGICAL HOSPITALS   683  
6293  REHABILITATION HOSPITALS (EXCEPT ALCOHOLISM, D...    80  
6661             GENERAL MEDICAL AND SURGICAL HOSPITALS   493  
6774             GENERAL MEDICAL AND SURGICAL HOSPITALS    23  
6777             GENERAL MEDICAL AND SURGICAL HOSPITALS   541  
6786             GENERAL MEDICAL AND SURGICAL HOSPITALS   126  
6808             GENERAL MEDICAL AND SURGICAL HOSPITALS   733  
6813             GENERAL MEDICAL AND SURGICAL HOSPITALS  1091

7. Which county in Tennessee has the most hospitals? There are 95 counties in TN. How many of them have no hospitals at all? Lengh = 80. 15 have no hospitals.

In [38]:
TN_hospitals["COUNTY"].value_counts()

SHELBY        23
DAVIDSON      15
HAMILTON      13
KNOX          11
WASHINGTON     6
              ..
HAYWOOD        1
OVERTON        1
CHEATHAM       1
DICKSON        1
ROBERTSON      1
Name: COUNTY, Length: 80, dtype: int64